src/Microsoft.OData.Core/ErrorUtils.cs

```diff
@@ -56,6 +57,25 @@ internal static void WriteXmlError(XmlWriter writer, ODataError error, bool incl
             WriteXmlError(writer, code, message, innerError, maxInnerErrorDepth);
         }
 
+        /// <summary>
+        /// Asynchronously Write an error message.
+        /// </summary>
+        /// <param name="writer">The Xml writer to write to.</param>
+        /// <param name="error">The error instance to write.</param>
+        /// <param name="includeDebugInformation">A flag indicating whether error details should be written (in debug mode only) or not.</param>
+        /// <param name="maxInnerErrorDepth">The maximum number of nested inner errors to allow.</param>
+        internal static async Task WriteXmlErrorAsync(XmlWriter writer, ODataError error, bool includeDebugInformation, int maxInnerErrorDepth)

```

Comment:
Do we still write the XML payload? 
I don't think so, we only output the JSON payload exception the CSDL, right?
If that's the case, why do we need to implement the XML related async methods?

Reply:
There are still some async tests and async methods that are calling this method.